src/main/java/org/commcare/cases/entity/NodeEntityFactory.java

```diff
@@ -73,10 +73,7 @@ public Detail getDetail() {
                  * We are wrapping the original exception in a new XPathException to avoid
                  * refactoring large number of functions caused by throwing XPathSyntaxException here.
                  */
-                XPathException xe = new XPathException(e.getMessage(), e);
-                if (e instanceof XPathException) {
-                    xe.setSource(((XPathException)e).getSource());
-                }
+                XPathException xe = new XPathException(e);

```

Comment:
Gotcha. I think I would still prefer if we only wrapped the `XPathSyntaxException`, since the only reason we are wrapping at all is to avoid the messy-ness of throwing a non-runtime exception. By wrapping both types of exceptions, it obfuscates our real reason for doing the wrapping, which I don't like.